pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -276,6 +278,10 @@ def __init__(
         super().__init__(
             parallel_devices=parallel_devices, num_nodes=num_nodes, cluster_environment=cluster_environment
         )
+
+        self.checkpoint_plugin = DeepSpeedPlugin() if checkpoint_plugin is None else checkpoint_plugin
+        self.checkpoint_plugin.training_type_plugin = self
```

Comment:
Very ugly to set like this. Eventually all plugins should expose the `checkpoint_plugin` to input this in. cc @ananthsub any ideas here?

Reply:
Actually may be wrong here. Might not need to expose `checkpoint_plugin` in the constructor of the training type plugin, and just need to set the checkpoint plugin. Maybe this can be done in the accelerator connector, with a default being set by the plugin itself!